bridge/svix-bridge-plugin-queue/src/receiver_output/mod.rs

```diff
@@ -44,7 +53,38 @@ impl ReceiverOutput for QueueForwarder {
     }
 
     async fn handle(&self, request: ForwardRequest) -> Result<(), BoxError> {
-        self.sender.send_serde_json(&request.payload).await?;
+        let mut sender = self.sender.lock().await;
+        // `QueueForwarder` initializes its sender lazily, but also the sender can be discarded if
+        // it gets in a bad state.
+        // When None, rebuild the sender.
+        if sender.is_none() {
+            *sender = Some(Self::build_sender(&self.opts).await?);
+        }
+
+        let res = sender
+            .as_ref()
+            .expect("non-none sender")
+            .send_serde_json(&request.payload)
+            .await;
+
+        // Certain clients (like rabbitmq) require intervention when the connection to the remote
+        // is interrupted. Check for those failure cases and invalidate the sender so it gets
+        // rebuilt next time.
+        //
+        // `QueueError::Generic` boxes the inner error so either we'd need to have a direct dep on
+        // `lapin` so we can downcast to inspect specifics, or otherwise we'd need omniqueue to
+        // re-export the error types.
+        //
+        // In practice multiple error types can show up that all mean the same thing so to start,
+        // just recycle for any rabbitmq error.
+        if let (Err(QueueError::Generic(_err)), QueueOutputOpts::RabbitMQ(_cfg)) =
+            (&res, &self.opts)
+        {
+            let _ = sender.take();
```

Comment:
Should we log the error here, so that we can tell if there's a bad configuration causing it to constantly reconnect?

Reply:
The error will get logged already on L86 (just below here). This is just checking to see if two things are true:

1. there was an error
2. we're talking to rabbit

... in which case we want to clear the sender so it can be rebuilt.